File: python/65d_1836A_Destroyer.py
def solve(arr, n):
    hmap = {}
    last = max(arr)
    for i in arr:
        if i in hmap:
            hmap[i] += 1
        else:
            hmap[i] = 1
    if 0 not in hmap:
        return "NO"
    #print(hmap)
    for i in range(1,last+1):
        if i not in hmap:
            return "NO"
        else:
            #print(i, hmap[i])
            if hmap[i] > hmap[i-1]:
                return "NO"
    return "YES"

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = list(map(int, input().split()))
        ans.append(solve(arr, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/65d_1836A_Destroyer.cpp
#include<bits/stdc++.h>

using namespace std;

string solve(map<int, int> hmap, int last){
    if(hmap[0] == 0){
        return "NO";
    }
    for(int i = 1; i <= last; i++){
        if(hmap[i] > hmap[i-1]){
            return "NO";
        }
    }
    return "YES";
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n, temp, last;
    vector<string> ans;
    cin>>t;

    for(int i = 0; i < t; i++){
        cin>>n;
        map<int, int> hmap;
        last = 0;
        for(int j = 0; j < n; j++){
            cin>>temp;
            hmap[temp]++;
            last = max(last, temp);
        }
        ans.push_back(solve(hmap, last));
    }

    for(auto i : ans){
        cout<<i<<"\n";
    }
    return 0;
}